coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+#Overview
+This code coverage tool has two major features.
+1. In post-submit workflow (when a pull request is merged), it runs code coverage and generates the 
+  following artifacts in a GCS bucket
+  - xml file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
+  - code coverage profile, which will be used later in presubmit workflow as a base for comparison
+2. In pre-submit workflow (when a pull requested is created or updated with new commit)
+  - it runs code coverage on target directories and compares the new result with the one stored by 
+  the post-submit workflow and generate coverage difference. It reports coverage changes to the 
+  pull request as a comment by a robot github account. 
+  - it uses go tools to generate line by line coverage and serve the result in html, with a link as 
+  part of the robot comment mentioned above 
+  - it can report a failure status if coverage falls below threshold - this allows repo owner to make the 
+check required and block PR if coverage threshold not met. 
+
+
+##Users
+The presubmit tool is intended for a developer to see the impact on code coverage of his/her commit. It can also be used by repo managers to block any PR from merging if the coverage falls under customized threshold.
```

Comment:
As above, this isn't provided by the Job. Your job doesn't have any logic related to merge automation, that is handled entirely by Tide. Your job just exists with a non-zero status if the line coverage percentage is below the configured threshold.

Reply:
This section is to describe who are the intended and potential users. Why should we limit the scope to what the tool can do standalone?

Reply:
This isn't a guide to all of our testing infra structure and how it can be used. This is a design document specifically for your coverage tool. If you include things like 
> It can also be used by repo managers to block any PR from merging if the coverage falls under customized threshold.

It really sounds like you are saying that your tool can do that on its own. Even if it was clear that Tide is providing this functionality, this isn't the place to describe that. As mentioned before, if you want to include use case specific recommendations please do so at the bottom of the document in a separate section or in a separate document altogether.